### ISSUE TITLE
Incorrect error message for borrow after closure capture

### ISSUE THREAD

**Issue by zwarich** [rust-lang/rust#14273]
The code

```rust
fn main() {
    let x = box 3;
    let f = proc() {
        let y = x;
    };
    let z = &x;
}
```

gives the error message

```
test-closure.rs:6:14: 6:15 error: use of moved value: `x`
test-closure.rs:6     let z = &x;
                               ^
test-closure.rs:3:13: 5:6 note: `x` moved here because it has type `proc():Send`, which is non-copyable (perhaps you meant to use clone()?)
test-closure.rs:3     let f = proc() {
test-closure.rs:4         let y = x;
test-closure.rs:5     };
error: aborting due to previous error
```

Obviously, `x` doesn't have the type `proc():Send`.

**Comment by steveklabnik** [comment]
This example has changed, but this error message is correct. The `box` moves into the closure, and so you can't use it after.
